DB-11382 make MON_GET_CONNECTION table valued function instead of system table

hatyo opened the pull request:
In this PR we change the definition of `MON_GET_CONNECTION` to become a table valued function similar to DB2's version. Previously `MON_GET_CONNECTION` was implemented as a system table, after some brainstorming we decided we shouldn't persist the data inside this table and therefore it should not be a system table.

Implementation wise, a new system VTI is added and aliased with `MON_GET_CONNECTION`, the views built on top it (`APPLICATIONS`, `SNAPAPPL_INFO`, `SNAPAPPL`) are rewritten to refer to the new table-valued function.

Now we can query this table valued function like this:

```
select * from table(sysibmadm.mon_get_connection()) x;
```
(we'll get empty results, as before).

ipraznik-splice commented on db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:
```diff
@@ -266,6 +266,14 @@ public void boot(boolean create,Properties startParams) throws StandardException
         //in this boot method.
+
+        try {
+            Thread.sleep(10 * 1000);
```
should Thread.sleep be removed?

hatyo replied:
Thanks, will do

hatyo commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

hatyo commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2456/

spliceci commented:
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2457/

jyuanca reviewed (changes requested):
Do you need to drop the table in the upgrade script?

I ran upgrade test from a build earlier than 1992, and got a failure when trying to run UpgradeScriptForTablePriorities.
021-02-16 21:29:00,934 (SpliceDatabaseLifecycleManager) WARN  [c.s.p.Exceptions] - Parsing exception, original stacktrace:
java.lang.NullPointerException
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.faultInTabInfo(DataDictionaryImpl.java:8027)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getNonCoreTI(DataDictionaryImpl.java:7839)
	at com.splicemachine.derby.impl.sql.catalog.SpliceDataDictionary.upgradeTablePriorities(SpliceDataDictionary.java:1511)
	at com.splicemachine.derby.impl.sql.catalog.upgrade.UpgradeScriptForTablePriorities.upgradeSystemTables(UpgradeScriptForTablePriorities.java:19)
	at com.splicemachine.derby.impl.sql.catalog.upgrade.UpgradeScriptBase.run(UpgradeScriptBase.java:61)
	at com.splicemachine.derby.impl.sql.catalog.upgrade.SpliceCatalogUpgradeScripts.runAllScripts(SpliceCatalogUpgradeScripts.java:145)
	at com.splicemachine.derby.impl.sql.catalog.upgrade.SpliceCatalogUpgradeScripts.runUpgrades(SpliceCatalogUpgradeScripts.java:159)
	at com.splicemachine.derby.impl.sql.catalog.SpliceDataDictionary.upgradeIfNecessary(SpliceDataDictionary.java:926)
	at com.splicemachine.derby.impl.sql.catalog.SpliceDataDictionary.loadDictionaryTables(SpliceDataDictionary.java:735)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.boot(DataDictionaryImpl.java:501)

Please do more upgrade test from a customer build, for example 1986 build for Cetera. Please try upgrading from 3.0 to 3.1 to master

hatyo commented:
@jyuanca thanks for the valuable feedback, there was indeed a problem with upgrade process which I fixed and verified by testing the following upgrade scenarios:
1982 -> DB-11382.
3.0 -> 3.1 -> DB-11382.
I also addressed merge conflicts.

hatyo commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2532/

hatyo commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST SUCCEEDED +1

carolp-503 reviewed (approved):
+1
